src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -493,7 +509,9 @@ impl<'a> Linker for GccLinker<'a> {
             if !self.is_ld {
```

Comment:
```suggestion
} else if !is_windows {
            if !self.is_ld {
```
`is_ld && is_windows` seems like a viable combination, but it will result in `-Wl,` without any linker option with the code as written.

Reply:
Actually, this should be exactly the combination that we get with `lld` (see `fn to_linker`)?

Reply:
I'm successfully linking Rust by passing `-fuse-ld=lld` to both GCC and Clang not (yet) using LLD on it's own. So we still need `arg.push("-Wl,")` for that case.

Reply:
Ah, ok, so we need `-Wl,foo.def` because `.def` files work without any prefix. Got it.